slash/react/src/Messages/Message.tsx

```diff
@@ -0,0 +1,133 @@
+   * If not provided, a default icon based on the classModifier will be used.
+   * <strong>This not recommended to use</strong>. Only use as an escape hatch.
+   */
+  icon?: string;
+  /**
+   * handler for the close button click event.
+   * If not provided, the close button will not be displayed.
+   */
+  onClose?: MouseEventHandler<HTMLButtonElement>;
+  /**
+   * If you want to add a custom class to the component.
+   * @example
+   * ```tsx
+   * <Message className="my-custom-class" title="My message">
+   * ```
+   * will produce
+   * ```html
+   * <div class="my-custom-class af-alert">My message</div>
+   */
+  className?: string;
+  /**
+   * @deprecated Use `variant` instead.
+   */
+  classModifier?: MessageVariants | "danger";
+  /**
+   * Variant of the message.
+   * This will determine the icon and the style of the message.
+   */
+  variant?: MessageVariants;
+  /**
+   * Aria label for the close button.
+   * This is important for accessibility, use a meaningful label to describe the action.
+   * @default "close"
+   */
+  closeButtonAriaLabel?: string;
+};
+
+const getIconUrl = (type: string) => {
+  switch (type) {
+    case "info":
+      return infoSvg;
+    case "success":
+      return successSvg;
+    case "error":
+      return errorSvg;
+    case "warning":
+      return warningSvg;
+    default:
+      return errorSvg;
+  }
+};
+
+const getVariant = (
+  classModifier: MessageVariants | "danger",
+  variant?: MessageVariants,
+) => {
+  if (variant) {
+    return variant;
+  }
+  if (classModifier) {
+    return classModifier === "danger" ? "warning" : classModifier;
+  }
+  return "error"; // Default variant
+};
+
+export const Message = ({
+  className,
+  onClose,
+  icon,
+  title,
+  children,
+  classModifier = "error",
+  variant, // Backward compatibility
+  closeButtonAriaLabel = "close",
+}: PropsWithChildren<MessageProps>) => {
+  const safeVariant = getVariant(classModifier, variant);
+
+  const componentClassName = getComponentClassNameWithUserClassname({
+    componentClassName: "af-alert",
+    userClassName: className,
+    classModifier: safeVariant,
+  });
+
+  const iconSrc = icon ?? getIconUrl(safeVariant);
+
+  return (
+    <div className={componentClassName} role="alert">
+      <div className="af-alert__sidebar">
+        <div className="af-alert__indicator">
+          <Svg src={iconSrc} className="af-alert__icon" role="presentation" />
+        </div>
+      </div>
+      <div className="af-alert__content">
+        <div className="af-alert__title">{title}</div>
+        {children}
+      </div>
+      {onClose ? (
+        <button
+          onClick={onClose}
+          aria-label={closeButtonAriaLabel}
+          type="button"
+          className="af-alert__close-button"
+        >
+          <Svg
+            src={closeSvg}
+            className="af-alert__close-icon"
+            role="presentation"
+          />
+        </button>
+      ) : null}
```

Comment:
Ce ne serait pas mieux si on fait comme ca ?

```suggestion
      {Boolean(onClose) && (
        <button
          onClick={onClose}
          aria-label={closeButtonAriaLabel}
          type="button"
          className="af-alert__close-button"
        >
          <Svg
            src={closeSvg}
            className="af-alert__close-icon"
            role="presentation"
          />
        </button>
      )}
```
Je comprend l'intégrés mais j'ai toujours un peut de mal avec le null.

il faudrait qu'on ce force à toujours encapsuler son condition dans le jsx d'un Boolean(xy)

Reply:
`null` et `false` sont identiques pour react, mais y a bien le warning dans la doc pour les `number` : https://react.dev/learn/conditional-rendering#logical-and-operator-

Par contre je suis pas d'accord avec toi, on devrait garder la bonne pratique de faire des ternaires, ca evite toute erreur.

Reply:
Oui je comprend parfaitement le pourquoi du ternaires surtout pour le number sring etc .. 
il est évident de ne pas mettre des éléments dans le dom qui n'aurait pas leur place.

cependant, la syntaxte ternaire dans le jsx n'est pas une pratique courante chez nous à axa, j'entends par la que beaucoup de dev on pris le reflexe (peut être à tord) de faire {foo && <div/>}

du coup histoire de pas trop changer leurs habitude faire un {Boolean(foo) && <div/>} me semble, peut être, être un bon compromis.

qu'en penses tu ?

PS: au pire on peux continuer ce débat en MP car pour moi c'est pas bloquant pour cette PR 😉